phf_shared/src/lib.rs

```diff
@@ -103,6 +106,14 @@ impl PhfHash for [u8] {
     }
 }
 
+#[cfg(feature = "unicase")]
+impl<S> PhfHash for unicase::UniCase<S>
+where unicase::UniCase<S>: Hash {
+    #[inline]
+    fn phf_hash<H: Hasher>(&self, state: &mut H) {
+        self.hash(state)
```

Comment:
I don't think so, it [hashes a char at a time](https://github.com/seanmonstar/unicase/blob/c09ee88f30ba6cf25d6ed23de6929987e9f245a1/src/lib.rs#L103-L105) (so no allocation required) and doesn't include the length. Unicase doesn't have any guarantee to keep this implementation, but I think it'd be breaking for them to change it in future so would be a version bump for them? cc @seanmonstar